Explorer/Assets/DCL/AvatarRendering/Emotes/Systems/Load/LoadEmotesByPointersSystem.cs

```diff
@@ -1,26 +1,281 @@
+        private readonly URLSubdirectory customStreamingSubdirectory;
+        private readonly URLBuilder urlBuilder = new ();
+
         public LoadEmotesByPointersSystem(
             World world,
             IWebRequestController webRequestController,
-            IStreamableCache<EmotesDTOList, GetEmotesByPointersFromRealmIntention> cache
+            IStreamableCache<EmotesDTOList, GetEmotesByPointersFromRealmIntention> cache,
+            IEmoteStorage emoteStorage,
+            IRealmData realmData,
+            URLSubdirectory customStreamingSubdirectory
         )
-            : base(world, cache, webRequestController) { }
+            : base(world, cache, webRequestController)
+        {
+            this.emoteStorage = emoteStorage;
+            this.realmData = realmData;
+            this.customStreamingSubdirectory = customStreamingSubdirectory;
+        }
 
         protected override EmotesDTOList CreateAssetFromListOfDTOs(RepoolableList<EmoteDTO> list) =>
             new (list);
+
+        protected override void Update(float t)
+        {
+            base.Update(t);
+
+            GetEmotesByPointersQuery(World, t);
+            DisposeFinishedIntentsQuery(World);
+        }
+
+        [Query]
+        [None(typeof(StreamableResult))]
+        private void GetEmotesByPointers([Data] float dt, in Entity entity,
+            ref GetEmotesByPointersIntention intention,
+            ref IPartitionComponent partitionComponent)
+        {
+            if (intention.TryCancelByRequest<GetEmotesByPointersIntention, EmotesResolution>(
+                    World!,
+                    GetReportCategory(),
+                    entity,
+                    static _ => "Pointer request cancelled"))
+                return;
+
+            using var _ = ListPool<URN>.Get(out var pointersToRequest)!;
+            var resolvedEmotesTmp = RepoolableList<IEmote>.NewList();
+
+            ExtractMissingPointersAndResolvedEmotes(in intention, pointersToRequest!, resolvedEmotesTmp);
+
+            if (intention.Timeout.IsTimeout(dt))
+            {
+                var pointersStrLog = string.Join(",", intention.Pointers);
+                ReportHub.LogWarning(GetReportCategory(), $"Loading emotes timed out, {pointersStrLog}");
+
+                ResolveIntentionWithSuccessfulEmotes(entity, intention, resolvedEmotesTmp);
+
+                return;
+            }
+
+            if (RequestMissingPointers(pointersToRequest!, partitionComponent, intention.BodyShape)) return;
+
+            bool success = GetAssetBundlesUntilAllAreResolved(in intention, partitionComponent, resolvedEmotesTmp.List);
+
+            if (success)
+                World!.Add(entity, NewEmotesResult(resolvedEmotesTmp, intention.Pointers.Count));
+        }
+
+        [Query]
+        [All(typeof(StreamableResult))]
+        private void DisposeFinishedIntents(in Entity entity, ref GetEmotesByPointersIntention intention)
+        {
+            intention.Dispose();
```

Comment:
We need to dispose the intention as it is no longer used, so we release the arrays from the pools. I didnt think of any other clean way of doing it. Any suggestion is welcome